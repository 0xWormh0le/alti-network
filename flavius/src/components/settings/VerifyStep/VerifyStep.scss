@import 'styles/_shared.scss';

.VerifyStep {
  &__title {
    color: $TEXT_COLOR;
    margin-bottom: 1rem;
  }

  &__form-group {
    width: 100px;
  }
}
