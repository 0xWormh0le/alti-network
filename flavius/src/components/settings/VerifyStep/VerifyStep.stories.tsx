import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import VerifyStep, { VerifyStepProps } from './VerifyStep'

export default {
  title: 'Settings/VerifyStep',
  component: VerifyStep
} as Meta

const Template: Story<VerifyStepProps> = (args) => <VerifyStep {...args} />

export const Default = Template.bind({})
Default.args = {}
