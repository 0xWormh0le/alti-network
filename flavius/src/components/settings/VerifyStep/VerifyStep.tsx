import React from 'react'

import Button from 'components/elements/Button'
import FormControl, { FormControlProps } from 'components/elements/FormControl'
import FormGroup from 'components/elements/FormGroup'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import UI_STRINGS from 'util/ui-strings'

import './VerifyStep.scss'

export interface VerifyStepProps {
  buttonLoading: boolean
  challengeAnswer: string
  onChallengeAnswerChange: (event: React.FormEvent<FormControlProps>) => void
  onStepCompleted: () => void
}

const VerifyStep: React.FC<VerifyStepProps> = ({
  buttonLoading,
  challengeAnswer,
  onChallengeAnswerChange,
  onStepCompleted
}) => (
  <section className='VerifyStep TwoFactorAuthentication__step'>
    <form
      onSubmit={(event) => {
        event.preventDefault()
        onStepCompleted()
      }}>
      <Typography variant={TypographyVariant.H3} weight='semibold' className='VerifyStep__title'>
        {UI_STRINGS.SETTINGS.TRY_YOUR_CODE}
      </Typography>
      <p>{UI_STRINGS.SETTINGS.USE_YOUR_SECURITY_APP_TO_GENERATE}</p>
      <FormGroup controlId='challengeAnswer' className='VerifyStep__form-group'>
        <FormControl autoFocus={true} value={challengeAnswer} onChange={onChallengeAnswerChange} />
      </FormGroup>
      <Button
        className='VerifyStep__button'
        disabled={challengeAnswer === ''}
        action='primary'
        text={UI_STRINGS.BUTTON_LABELS.VERIFY}
        loadingText={UI_STRINGS.SETTINGS.VERIFYING}
        isLoading={buttonLoading}
        onClick={onStepCompleted}
      />
    </form>
  </section>
)

export default VerifyStep
