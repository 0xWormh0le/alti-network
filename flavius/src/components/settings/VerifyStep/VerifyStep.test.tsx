import React from 'react'
import { render } from '@testing-library/react'

import noop from 'lodash/noop'
import VerifyStep from './VerifyStep'

describe('VerifyStep', () => {
  it('renders correctly', () => {
    const localProps = {
      buttonLoading: false,
      challengeAnswer: 'answer',
      onChallengeAnswerChange: noop,
      onStepCompleted: noop
    }
    const { container } = render(<VerifyStep {...localProps} />)
    expect(container).toMatchSnapshot()
  })
})
