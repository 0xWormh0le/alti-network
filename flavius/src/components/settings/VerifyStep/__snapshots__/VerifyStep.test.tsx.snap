// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyStep renders correctly 1`] = `
<div>
  <section
    class="VerifyStep TwoFactorAuthentication__step"
  >
    <form>
      <h3
        class="Typography Typography--h3 Typography--semibold VerifyStep__title"
      >
        Try your code
      </h3>
      <p>
        Use your security app to generate a code and enter it below.
      </p>
      <div
        class="FormGroup VerifyStep__form-group"
      >
        <input
          class="FormControl FormControl--flat"
          id="challengeAnswer"
          value="answer"
        />
      </div>
      <button
        class="Button VerifyStep__button Button--primary"
        type="button"
      >
        Verify
      </button>
    </form>
  </section>
</div>
`;
